### ISSUE TITLE
Non required array is not marked as omitempty

### ISSUE THREAD

**Issue by mhmxs** [go-swagger/go-swagger#1206]
## Problem statement

Our endpoint receives an optional array.
```
        "recipeIds" : {
          "type" : "array",
          "description" : "referenced recipe ids",
          "uniqueItems" : true,
          "items" : {
            "type" : "integer",
            "format" : "int64"
          }
        },
```

And in the generated source i see:
```
RecipeIds []int64 `json:"recipeIds"`
```

I would like to ask what is the reason to not mark the array as omitempty? Because the API allows to not send this data, and it allows to send an array as data also, but in this case go-swagger sends a `null`. I have two issues with `null`, first some API doesn't handle `null` as well (i know it is an API issue, but there are APIs out of my scope and also legacy APIs) and the other is if i generate a minimal JSON template for my users it will be full with unnecessary `null` values which looks not so nice.

## Environment
swagger version: 0.12.0  
go version: 1.9.1

**Comment by casualjim** [comment]
I just merged x-omitempty (#1189)
if you add that to the spec you can skip then nulls

**Comment by mhmxs** [comment]
Thanks a lot.
